doc/manuals/sprokit/embedded_pipeline.rst

```diff
@@ -0,0 +1,327 @@
+*relativepath* modifiers can supply an accurate prefix. If the
+pipeline comes from a string stream, do not supply this parameter.
+
+After the pipeline has been built, it is started with the ``start()``
+call. Data are supplied to the pipeline via the ``send()`` method and
+retrieved with the ``receive()`` method. If input data is supplied
+faster than the pipeline can accept it, the ``send()`` method will
+block until there is room for the new element. Similarly, the
+``receive()`` method will block if there is no output available from the
+pipeline.
+
+When all the input has been supplied to the pipeline, the
+``send_end_if_input()`` is used to signal this condition. This will
+cause the pipeline terminate after all the supplied data has been
+processed. The ``receive()`` method will return an end of data
+indication after the last pipeline output has been returned.
+
+Example
+-------
+
+Lets look at an example to see the details of using an embedded
+pipeline. The following code implements a simple pipeline that just
+passes data from the input to the output. The following sections will
+describe sections of this example indetail.::
+
+  #include <sprokit/pipeline_util/literal_pipeline.h>
+
+  // SPROKIT macros can be used to create pipeline description
+  std::stringstream pipeline_desc;
+  pipeline_desc << SPROKIT_PROCESS( "input_adapter",  "ia" )
+                << SPROKIT_PROCESS( "output_adapter", "oa" )
+
+                << SPROKIT_CONNECT( "ia", "counter",  "oa", "out_num" )
+                << SPROKIT_CONNECT( "ia", "port2",    "oa", "port3" )
+                << SPROKIT_CONNECT( "ia", "port3",    "oa", "port2" )
+    ;
+
+  // create embedded pipeline
+  kwiver::embedded_pipeline ep;
+  ep.build_pipeline( pipeline_desc );
+
+  // Start pipeline
+  ep.start();
+
+  for ( int i = 0; i < 10; ++i)
+  {
+    // Create dataset for input
+    auto ds = kwiver::adapter::adapter_data_set::create();
+
+    // Add value to be pushed to the named port
+    ds.add_value( "counter", i );
+
+    // Data values need to be supplied to all connected ports
+    // (based on the previous pipeline definition)
+    ds.add_value( "port2", i );
+    ds.add_value( "port3", i );
+    ep.send( ds ); // push into pipeline
+
+    // Get output from pipeline
+    auto rds = ep.receive();
+
+    // get value from the output adapter
+    int val = get_port_data<int>( "out_num" );
+  } // end for
+
+  ep.send_end_of_input(); // indicate end of input
+
+  auto rds = ep.receive(); // Retrieve end of input data item.
+  if ( ! ep.at_end() || ! rds.is_end_of_data() )
+  {
+    // This is unexpected. Both conditions should be true.
+  }
+
+
+The following section builds a pipeline at runtime using a set of
+macros to simplify constructing the pipeline definition. The full list
+of these macros and associated documentation can be found in the
+documentation for the include file. ::
+
+    1)    #include <sprokit/pipeline_util/literal_pipeline.h>
```

Comment:
The manual line numbers here feel a little awkward and disrupt the ability to apply syntax highlighting; I would consider instead using:

    .. code:: c++
       :number-lines:

(...or maybe [code-block](http://www.sphinx-doc.org/en/stable/markup/code.html), though I'm not sure if or why sphinx's version is "better".)